tests/dataframe_test.py

```diff
@@ -1442,5 +1437,667 @@ def test_to_markdown(self):
+        
+        # string index
+        df3 = ak.DataFrame({"ints": ak.array(ints), "floats":ak.array(floats), "strings":ak.array(strings)},index=string_index)
+        _df3 = pd.DataFrame({"ints": np.array(ints), "floats":np.array(floats), "strings":np.array(strings)},index=string_index)
+
+
+         #group 2: int labels
+
+        # default index
+        df4 = ak.DataFrame({1: ak.array(ints), 2:ak.array(floats), 3:ak.array(strings)})
+        _df4 = pd.DataFrame({1: np.array(ints), 2:np.array(floats), 3:np.array(strings)})
+
+        # unorderd index
+        df5 = ak.DataFrame({1: ak.array(ints), 2:ak.array(floats), 3:ak.array(strings)},index=unordered_index)
+        _df5 = pd.DataFrame({1: np.array(ints), 2:np.array(floats), 3:np.array(strings)},index=unordered_index)
+        
+        # string index
+        df6 = ak.DataFrame({1: ak.array(ints), 2:ak.array(floats), 3:ak.array(strings)},index=string_index)
+        _df6 = pd.DataFrame({1: np.array(ints), 2:np.array(floats), 3:np.array(strings)},index=string_index)
+
+        access = _df4[1]
+
+        string_keys = ['ints', 'floats', 'strings']
+        int_keys = [1,2,3]
+
+        dfs = [df1,df2,df3,df4,df5,df6]
+        _dfs = [_df1,_df2,_df3,_df4,_df5,_df6]
+        keys_list = [string_keys, string_keys, string_keys, int_keys, int_keys, int_keys]
+        indexes = [default_index,unordered_index,string_index,default_index,unordered_index,string_index]
+        for (df,_df,keys,index) in zip(dfs, _dfs, keys_list, indexes):
+            # single column label returns a series
+            for key in keys:
+                access1_ = _df[key]
+                access1 = df[key]
+                self.assertIsInstance(access1_, pd.Series)
+                self.assertIsInstance(access1, ak.Series)
+                self.assertListEqual(access1_.values.tolist(), access1.values.to_list())
+                self.assertListEqual(access1_.index.tolist(), access1.index.to_list())
+            
+            # matching behavior for nonexistant label
+            with self.assertRaises(KeyError):
+                _access2 = _df[keys[0] * 100]
+            with self.assertRaises(KeyError):
+                access2 = df[keys[0] * 100]
+
+            # result reference behavior
+            _access3 = _df[keys[0]]
+            access3 = df[keys[0]]
+            access3[index[0]] = 100
+            _access3[index[0]] = 100
+            self.assertEqual(_df[keys[0]][index[0]], df[keys[0]][index[0]])
+
+            # key type matches column label types
+            with self.assertRaises(TypeError):
+                if isinstance(keys[0], int):
+                    a = df['int']
+                else:
+                    a = df[3]
+            with self.assertRaises(TypeError):
+                b = df[1.0]
+
+        # slices
+        def slice_check(_slice_access, slice_access):
```

Comment:
I'm not going to comment on every instance, but in general, these checks can be more succinct using a combination of `.to_pandas`, `assert_frame_equal`, and `assert_series_equal`.